src/helium_positioning_api/Models.py

```diff
@@ -85,4 +91,61 @@ def predict(self, uuid: str) -> Prediction:
             midpoint_lat, midpoint_long = midpoint(
                 sorted_hotspots[0], sorted_hotspots[1]
             )
+        else:
+            logger.warning(
+                "Not enough hotspots to perform Midpoint approximation."
+                "Using nearest neighbor model instead."
+            )
         return Prediction(uuid=uuid, lat=midpoint_lat, lng=midpoint_long)
+
+
+class Trilateration(Model):
+    """This model predicts the location of a given device using trilateration."""
+
+    def predict(self, uuid: str, model: str) -> Prediction:
+        """Create an object of Class Prediction.
+
+        :param uuid: Device id
+        :param model: Model to use for distance prediction
+
+        :return: coordinates of predicted location
+        """
+        model = get_model(model)
+        hotspots = self.get_hotspots(uuid)
+        sorted_hotspots = sorted(hotspots, key=lambda h: h.rssi)
+
+        if len(sorted_hotspots) < 3:
+            logger.warning(
+                "Not enough hotspots to perform trilateration. "
+                "Using nearest neighbor model instead."
+            )
+            return NearestNeighborModel().predict(uuid)
+
+        distance, longitude, latitude = [], [], []
+        for hotspot in sorted_hotspots:
+            dist = predict_distance(
+                model,
+                [
+                    hotspot.lat,
+                    hotspot.lng,
+                    hotspot.rssi,
+                    hotspot.snr,
+                    hotspot.spreading,
+                ],
+            )
+            longitude.append(hotspot.lng)
+            latitude.append(hotspot.lat)
+            distance.append(dist)
+
+        hotspots = zip(latitude, longitude, distance, strict=False)
+
+        # [[lat, long, dist], [lat, long, dist], [lat, long, dist]]
+
+        # do trilateration
```

Comment:
At least trilateration would be nice to have it as part of this PR.

Reply:
We planned adding trilateration in a new branch but will do it here.
Setting PR to draft for now